config/config.go

```diff
@@ -324,6 +325,37 @@ func (cfg BaseConfig) ValidateBasic() error {
 	default:
 		return errors.New("unknown log_format (must be 'plain' or 'json')")
 	}
+
+	switch cfg.ProxyApp {
+	case "kvstore",
+		"kvstore_connsync",
+		"kvstore_unsync",
+		"persistent_kvstore",
+		"persistent_kvstore_connsync",
+		"persistent_kvstore_unsync",
+		"e2e",
+		"e2e_connsync",
+		"e2e_unsync",
+		"noop":
+	default: // network address
+		parts := strings.SplitN(cfg.ProxyApp, "://", 2)
+		if len(parts) != 2 { // TCP address
+			_, err := net.ResolveTCPAddr("tcp", cfg.ProxyApp)
+			if err != nil {
+				return fmt.Errorf("failed to resolve TCP proxy_app %s: %w", cfg.ProxyApp, err)
+			}
+		} else {
+			switch parts[0] { // other protocol
```

Comment:
here's the validation is only in the protocol (`parts[0]`) but would it be possible to have a validation on the address (`parts[1]`) ? at least from a well form perspective ?